packages/fx-core/src/component/generator/spfx/depsChecker/yoChecker.ts

```diff
@@ -193,7 +194,7 @@ export class YoChecker implements DependencyChecker {
       await fs.ensureFile(this.getSentinelPath());
     } catch (error) {
       void this._logger.error(`Failed to execute npm install ${displayName}@${version}`);
-      throw NpmInstallError(error as Error);
+      throw new NpmInstallError(error as Error);
```

Comment:
Moving common errors to one file is good practice, but should we set correct source so we can better identify where the error comes from?

Reply:
Yes, I will create a new task to fix this issue.